From 4d22e078c21eaab501ee6333f3d2d34aab5839c4 Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Tue, 30 Mar 2021 13:13:47 +0000
Subject: [PATCH 1044/1077] MANTIS 0060035 CLA - Support biometric
 identification

add CUSS-APP-Transfer-OneID.xsd

CUSS-1.5.BIOMETRICS.GP.XSD
add gp-parameter-name
- materna.env.clientId.airport
- materna.env.clientId.flightOperatingCarrier
- materna.quality.maskDetected
- materna.quality.fakeImageDetected
- materna.cuss.identify.carrierCodeList
- materna.cbp.tvs.password
- materna.image.width
- materna.image.height
- materna.image.sharpness
- materna.image.temperature
- materna.image.imageTemplate
- materna.kpi.impTxDuration

git-svn-id: svn://localhost/SelfServiceCommon/trunk@1644 90b65887-3827-0410-9a23-83215b262276
---
 .../idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD       |  59 +++++++-
 .../idl/Cuss/CUSS-APP-Transfer-OneID.xsd      | 134 ++++++++++++++++++
 2 files changed, 192 insertions(+), 1 deletion(-)
 create mode 100644 SelfServiceCommon/Massai/idl/Cuss/CUSS-APP-Transfer-OneID.xsd

diff --git a/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD b/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD
index 4ac9d993..6eb77c23 100644
--- a/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD
+++ b/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD
@@ -19,6 +19,16 @@
           <xs:documentation>The (kiosk) name property from the client ID</xs:documentation>
         </xs:annotation>
       </xs:enumeration>
+      <xs:enumeration value="materna.env.clientId.airport">
+        <xs:annotation>
+          <xs:documentation>The airport property from the client ID</xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <xs:enumeration value="materna.env.clientId.flightOperatingCarrier">
+        <xs:annotation>
+          <xs:documentation>The flight operating carrier property from the client ID</xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
       <!-- Namespace 'materna.quality' -->
       <xs:enumeration value="materna.quality.verticalPosition">
         <xs:annotation>
@@ -44,7 +54,7 @@
         <xs:annotation>
           <xs:documentation>
             <![CDATA[Mimumim required FrontalFace quality for capturing facial images. Also used in the oposite direction for indicating the FrontalFace quality to the application]]>
-</xs:documentation>
+          </xs:documentation>
         </xs:annotation>
       </xs:enumeration>
       <xs:enumeration value="materna.quality.brightness">
@@ -122,6 +132,16 @@
           <xs:documentation>Absolute calculated/required image quality. Only pictures above this value must be sent by the device and processed by KA</xs:documentation>
         </xs:annotation>
       </xs:enumeration>
+      <xs:enumeration value="materna.quality.maskDetected">
+        <xs:annotation>
+          <xs:documentation>-1: not applicable, 0: medium applicable, 1: fully applicable. No explicit weight value required, since only used as indicator to the application</xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <xs:enumeration value="materna.quality.fakeImageDetected">
+        <xs:annotation>
+          <xs:documentation>-1: not applicable, 0: medium applicable, 1: fully applicable. No explicit weight value required, since only used as indicator to the application</xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
       <xs:enumeration value="materna.quality.verticalPosition.weight">
         <xs:annotation>
           <xs:documentation></xs:documentation>
@@ -249,12 +269,49 @@
           <xs:documentation>Enable (value 'true') the camera flash (if available and activated) while taking pictures. This improves the light conditions and passenger attention.</xs:documentation>
         </xs:annotation>
       </xs:enumeration>
+      <xs:enumeration value="materna.cuss.identify.carrierCodeList">
+        <xs:annotation>
+          <xs:documentation>Optional list of carrier codes for biometric identification via a connected IMP. Required if IMP does not support 'carrier code'-less identification. Format: list of comma separated carrier-codes (e.g. 'DL,NH')</xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
       <!-- Namespace 'materna.cbp'> -->
       <xs:enumeration value="materna.cbp.tvs.password">
         <xs:annotation>
           <xs:documentation>The password for the CBP TVS (Traveller Verification System). Required due to TVS password policy (change every 90 days) but service handler uses local static properties file</xs:documentation>
         </xs:annotation>
       </xs:enumeration>
+      <!-- Namespace 'materna.image'. Image related data -->
+      <xs:enumeration value="materna.image.width">
+        <xs:annotation>
+          <xs:documentation>Width of the image in pixels if available</xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <xs:enumeration value="materna.image.height">
+        <xs:annotation>
+          <xs:documentation>Height of the image in pixels if available</xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <xs:enumeration value="materna.image.sharpness">
+        <xs:annotation>
+          <xs:documentation>Sharpness of the image in percent if available</xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <xs:enumeration value="materna.image.temperature">
+        <xs:annotation>
+          <xs:documentation>Passenger temperature in degress centigrade if available</xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <xs:enumeration value="materna.image.imageTemplate">
+        <xs:annotation>
+          <xs:documentation>Face image template if available (e.g. 256 byte extract of the ID3 algorithm)</xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <!-- Namespace 'materna.kpi'. Key performance indices -->
+      <xs:enumeration value="materna.kpi.impTxDuration">
+        <xs:annotation>
+          <xs:documentation>How long the related IMP tranction (tx) needed (ms)</xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
     </xs:restriction>
   </xs:simpleType>
   <xs:simpleType name="association-data-tag">
diff --git a/SelfServiceCommon/Massai/idl/Cuss/CUSS-APP-Transfer-OneID.xsd b/SelfServiceCommon/Massai/idl/Cuss/CUSS-APP-Transfer-OneID.xsd
new file mode 100644
index 00000000..4ada5acb
--- /dev/null
+++ b/SelfServiceCommon/Massai/idl/Cuss/CUSS-APP-Transfer-OneID.xsd
@@ -0,0 +1,134 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Mit XMLSpy v2020 (x64) (http://www.altova.com) von Materna Information & Communications SE (Materna Information & Communications SE) bearbeitet -->
+<xs:schema xmlns:oneid="http://www.materna.com/CUSS/AppTransferOneID-1.0" xmlns:oneIdTypes="http://www.materna.com/CUSS/OneIdTypes-1.0" xmlns:xs="http://www.w3.org/2001/XMLSchema" targetNamespace="http://www.materna.com/CUSS/AppTransferOneID-1.0" elementFormDefault="qualified" attributeFormDefault="unqualified">
+	<xs:import namespace="http://www.materna.com/CUSS/OneIdTypes-1.0" schemaLocation="OneIdTypes.xsd"/>
+	<xs:complexType name="version">
+		<xs:annotation>
+			<xs:documentation>schema version; increase fixed values of version when schema is changed</xs:documentation>
+		</xs:annotation>
+		<xs:sequence>
+			<xs:element name="majorVersion" type="xs:int" fixed="1">
+				<xs:annotation>
+					<xs:documentation>major version</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="minorVersion" type="xs:int" fixed="1">
+				<xs:annotation>
+					<xs:documentation>minor version</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="revision" type="xs:int" fixed="0">
+				<xs:annotation>
+					<xs:documentation>revision</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+		</xs:sequence>
+	</xs:complexType>
+	<xs:element name="gp-data-item">
+		<xs:annotation>
+			<xs:documentation>General purpose biometric function request and/or response individual data item.</xs:documentation>
+		</xs:annotation>
+		<xs:complexType>
+			<xs:simpleContent>
+				<xs:extension base="xs:string">
+					<xs:attribute name="data-item-name" type="xs:string" use="required"/>
+				</xs:extension>
+			</xs:simpleContent>
+		</xs:complexType>
+	</xs:element>
+	<xs:element name="gp-data-list">
+		<xs:annotation>
+			<xs:documentation>General purpose biometric function request and/or response data item list.</xs:documentation>
+		</xs:annotation>
+		<xs:complexType>
+			<xs:sequence>
+				<xs:element ref="oneid:gp-data-item" maxOccurs="unbounded"/>
+			</xs:sequence>
+		</xs:complexType>
+	</xs:element>
+	<xs:element name="OneIDAppTransfer">
+		<xs:annotation>
+			<xs:documentation>OneID Application Data for the CUSS Application Transfer from the OneID Application to SBD/CKI or vice versa.</xs:documentation>
+		</xs:annotation>
+		<xs:complexType>
+			<xs:sequence>
+				<xs:element name="interfaceVersion" type="oneid:version">
+					<xs:annotation>
+						<xs:documentation>Version of the OneID Application Data  for the CUSS Application Transfer (i.e. XML Schema version).</xs:documentation>
+					</xs:annotation>
+				</xs:element>
+				<xs:element name="correlationId" type="xs:string">
+					<xs:annotation>
+						<xs:documentation>Application cross-sessional correlation id. Must be assigned by the first application in the application 'queue'</xs:documentation>
+					</xs:annotation>
+				</xs:element>
+				<xs:element name="language" type="xs:string" minOccurs="0">
+					<xs:annotation>
+						<xs:documentation>Language Identifier of current language in the calling application. - See http://www.ietf.org/rfc/rfc3066.txt</xs:documentation>
+					</xs:annotation>
+				</xs:element>
+				<xs:element name="passenger" type="oneIdTypes:passenger" maxOccurs="unbounded"/>
+				<xs:element name="applications" minOccurs="0">
+					<xs:annotation>
+						<xs:documentation>Specify source and target for the CUSS application transfer and the options for these applications.</xs:documentation>
+					</xs:annotation>
+					<xs:complexType>
+						<xs:sequence>
+							<xs:element name="source" type="xs:string" minOccurs="0">
+								<xs:annotation>
+									<xs:documentation>Calling application. - Use full CUSS application identifier.</xs:documentation>
+								</xs:annotation>
+							</xs:element>
+							<xs:element name="target" minOccurs="0">
+								<xs:annotation>
+									<xs:documentation>Receiving application. - Use full CUSS application identifier.</xs:documentation>
+								</xs:annotation>
+								<xs:complexType>
+									<xs:sequence>
+										<xs:element name="functions" minOccurs="0" maxOccurs="unbounded">
+											<xs:annotation>
+												<xs:documentation>What type of functions to be executed (associate, enrol, identify, verify, ...) Pls. check enumeration.</xs:documentation>
+											</xs:annotation>
+											<xs:simpleType>
+												<xs:restriction base="xs:string">
+													<xs:enumeration value="snapshot"/>
+													<xs:enumeration value="associate"/>
+													<xs:enumeration value="disassociate"/>
+													<xs:enumeration value="enrol"/>
+													<xs:enumeration value="delete"/>
+													<xs:enumeration value="identify"/>
+													<xs:enumeration value="verify"/>
+												</xs:restriction>
+											</xs:simpleType>
+										</xs:element>
+										<xs:element name="showDangerousGoods" type="xs:boolean" minOccurs="0">
+											<xs:annotation>
+												<xs:documentation>Specify whether the Dangerous Goods screen needs to be displayed by the callee.</xs:documentation>
+											</xs:annotation>
+										</xs:element>
+										<xs:element name="embeddedInvocation" type="xs:boolean" minOccurs="0">
+											<xs:annotation>
+												<xs:documentation>Indicates if application was called embedded in a different BL flow (e.g. if SBD requires ID-Check)</xs:documentation>
+											</xs:annotation>
+										</xs:element>
+										<xs:element name="accessibilityMode" type="xs:boolean" minOccurs="0">
+											<xs:annotation>
+												<xs:documentation>Indicates if application was called with enabled active accessibility mode (ADA). Target application has to activate the accessibility mode if required</xs:documentation>
+											</xs:annotation>
+										</xs:element>
+									</xs:sequence>
+								</xs:complexType>
+							</xs:element>
+						</xs:sequence>
+					</xs:complexType>
+				</xs:element>
+				<xs:element ref="oneid:gp-data-list" minOccurs="0"/>
+				<xs:element name="overallResult" type="oneIdTypes:resultCode" minOccurs="0">
+					<xs:annotation>
+						<xs:documentation>Overall result of the performed biometrics action. See passenger results for more details</xs:documentation>
+					</xs:annotation>
+				</xs:element>
+			</xs:sequence>
+		</xs:complexType>
+	</xs:element>
+</xs:schema>
-- 
2.41.0.windows.1

